services/productService.js

```diff
@@ -0,0 +1,29 @@
+const productDao = require("../models/productDao.js");
+
+const allProducts = async (page, pageSize, category, sub, type, sort) => {
+  let start = 0;
+  if (page <= 0) {
+    page = 1;
+  } else {
+    start = (page - 1) * pageSize;
+  }
+  const moreThanProducts = await productDao.moreThanProducts();
+  if (page > Math.ceil(moreThanProducts[0]["count(*)"] / pageSize)) {
+    const err = new Error("제품이 없습니다.");
+    err.statusCode = 400;
+    throw err;
+  }
```

Comment:
Dao 함수명 수정에 따라 해당 부분도 변수, 함수명 수정해주세요~!
```suggestion
  const productCounts = await productDao.getProductsCounts();
  if (page > Math.ceil(productCounts.counts / pageSize)) {
    const err = new Error("제품이 없습니다.");
    err.statusCode = 400;
    throw err;
  }
```